src/main/java/com/dawkastresu/medicalclinic/PatientService.java

```diff
@@ -0,0 +1,72 @@
+package com.dawkastresu.medicalclinic;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Objects;
+
+//ReauiredArgsConstruktor generuje konstruktor który przyjmuje pola finalne
+@RequiredArgsConstructor
+//Adnotacja Service mówi że ta klasa jest beanem zawierajacym logike biznesową aplikacji
+@Service
+public class PatientService {
+    //PatientRepository również musi być beanem aby Spring mógł dostarczyć instancję tej klasy
+    private final PatientRepository patientRepository;
+
+    public void editByEmail(String email, Patient newPatient) {
+        PatientValidator.newValueNotNullValidate(newPatient);
+        PatientValidator.validatePatientEdit(newPatient);
+        Patient patient = patientRepository.findByEmail(email)
+                .orElseThrow(() -> new PatientNotFoundException("Patient not found", HttpStatus.NOT_FOUND));
+
+        PatientValidator.cardIdNrNotChangedValidate(patient, newPatient);
+
+        patient.setEmail(newPatient.getEmail());
+        patient.setPassword(newPatient.getPassword());
+        patient.setFirstName(newPatient.getFirstName());
+        patient.setLastName(newPatient.getLastName());
+        patient.setPhoneNumber(newPatient.getPhoneNumber());
+        patient.setBirthday(newPatient.getBirthday());
+    }
+
+    public void editPasswordByMail(String email, PatientPassword password) {
+        Patient patient = patientRepository.findByEmail(email)
+                .orElseThrow(() -> new PatientNotFoundException("Patient not found", HttpStatus.NOT_FOUND));
+        patient.setPassword(password.getPassword());
+    }
+
+    public List<PatientDto> getAll() {
+        return patientRepository.findAll().stream()
+                .map(this::mapToDto)
+                .toList();
+    }
+
+    public PatientDto findPatientByName(String email) {
+        return patientRepository.findByEmail(email)
+                .map(this::mapToDto)
+                .orElseThrow(() -> new PatientNotFoundException("Patient not found", HttpStatus.NOT_FOUND));
+    }
+
+    public PatientDto addNew(Patient patient) {
+        PatientValidator.validatePatient(patient, patientRepository);
+        patientRepository.add(patient);
+        return mapToDto(patient);
+    }
+
+    public void removeByMail(String email) {
+        patientRepository.removeByEmail(email);
+    }
+
+    private PatientDto mapToDto(Patient patient) {
```

Comment:
Lepiej by bylo utworzyć klasę PatientMapper w niej mozes zrobić np. metody statyczne które będą to robić